scopus-email-file-downloader/src/main/java/no/sikt/nva/scopus/email/ScopusEmailHandler.java

```diff
@@ -0,0 +1,139 @@
+    public static final String ERROR_UNEXPECTED_STATUS_CODE = "Unexpected status code '%s' for request to uri";
+    public static final String REGEX_PATTERN_VALID_URL = "https://(sccontent-scudd-delivery-prod\\.s3\\.amazonaws\\.com"
+            + "/sccontent-scudd-delivery-prod/|localhost){1}[\\w.\\-/:#?=&;%~+]+";
+    private final S3Client s3Client;
+    private final HttpClient httpClient;
+
+    @JacocoGenerated
+    public ScopusEmailHandler() {
+        this(S3Driver.defaultS3Client().build(), HttpClient.newBuilder().build());
+    }
+
+    public ScopusEmailHandler(S3Client s3Client, HttpClient httpClient) {
+        this.s3Client = s3Client;
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public Void handleRequest(S3Event event, Context context) {
+        return attempt(() -> readFile(event))
+                .map(this::processEmailFile)
+                .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+    }
+
+    private String readFile(S3Event event) {
+        var s3Driver = new S3Driver(s3Client, extractBucketName(event));
+        var fileUri = createS3BucketUri(event);
+        return s3Driver.getFile(UriWrapper.fromUri(fileUri).toS3bucketPath());
+    }
+
+    private String extractBucketName(S3Event event) {
+        return event.getRecords().get(SINGLE_EXPECTED_RECORD).getS3().getBucket().getName();
+    }
+
+    private String extractFilename(S3Event event) {
+        return event.getRecords().get(SINGLE_EXPECTED_RECORD).getS3().getObject().getKey();
+    }
+
+    private URI createS3BucketUri(S3Event s3Event) {
+        return URI.create(String.format(S3_URI_TEMPLATE, extractBucketName(s3Event), extractFilename(s3Event)));
+    }
+
+    private Void processEmailFile(String emailContents) {
+        Email email = EmailConverter.emlToEmailBuilder(emailContents).buildEmail();
+        extractValidUrlsFromEmail(email.getPlainText()).forEach(this::downloadFile);
+        return null;
+    }
+
+    private Set<URI> extractValidUrlsFromEmail(String emailContents) {
+        Set<URI> uris = new HashSet<>();
+        Pattern pattern = Pattern.compile(REGEX_PATTERN_VALID_URL, Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(emailContents);
+        while (matcher.find()) {
+            uris.add(URI.create(emailContents.substring(matcher.start(0),matcher.end(0))));
+        }
+        return uris;
+    }
+
+    private void downloadFile(URI uri) {
+        String filename = extractFilenameFromUri(uri);
+        HttpRequest request = HttpRequest.newBuilder().uri(uri).build();
+        try {
+            HttpResponse<InputStream> httpResponse =
+                    httpClient.send(request, HttpResponse.BodyHandlers.ofInputStream());
+            if (httpResponse.statusCode() == OK) {
+                s3Client.putObject(newPutObjectRequest(filename), createRequestBody(httpResponse.body()));
```

Comment:
Could you add the functionality of sending bytes and not string to the S3Driver?